state/writer.go

```diff
@@ -0,0 +1,105 @@
+package state
+
+import (
+	"io"
+
+	"github.com/cycloidio/terracognita/errcode"
+	"github.com/cycloidio/terracognita/provider"
+	"github.com/hashicorp/terraform/addrs"
+	"github.com/hashicorp/terraform/states"
+	"github.com/hashicorp/terraform/states/statefile"
+	"github.com/hashicorp/terraform/states/statemgr"
+	"github.com/pkg/errors"
+)
+
+// Writer is a Writer implementation
+// that is meant to generate a TFState
+type Writer struct {
+	Config map[string]provider.Resource
+	writer io.Writer
+	state  *states.SyncState
+}
+
+// NewWriter returns a TFStateWriter initialization
+func NewWriter(w io.Writer) *Writer {
+	return &Writer{
+		Config: make(map[string]provider.Resource),
+		writer: w,
+		state:  states.NewState().SyncWrapper(),
+	}
+}
+
+// Write expects a key similar to "aws_instance" and the value to be *terraform.ResourceState
+// repeated keys will report an error
+func (w *Writer) Write(key string, value interface{}) error {
+	if key == "" {
+		return errcode.ErrWriterRequiredKey
+	}
+
+	if value == nil {
+		return errcode.ErrWriterRequiredValue
+	}
+
+	if _, ok := w.Config[key]; ok {
+		return errors.Wrapf(errcode.ErrWriterAlreadyExistsKey, "with key %q", key)
+	}
+
+	r, ok := value.(provider.Resource)
+	if !ok {
+		return errors.Wrapf(errcode.ErrWriterInvalidTypeValue, "expected provider.Resource, found %T", value)
+	}
+
+	absAddr := addrs.AbsResourceInstance{
+		Module: nil,
+		Resource: addrs.ResourceInstance{
+			Resource: addrs.Resource{
+				Mode: addrs.ManagedResourceMode,
+				Type: r.Type(),
+				Name: key,
+			},
+			Key: nil,
+		},
+	}
+
+	absProviderConf := addrs.AbsProviderConfig{
+		Module: nil,
+		ProviderConfig: addrs.ProviderConfig{
+			Type: r.Provider().String(),
+		},
+	}
+
+	src, err := r.ResourceInstanceObject().Encode(r.CoreConfigSchema().ImpliedType(), uint64(r.TFResource().SchemaVersion))
+	if err != nil {
+		return err
+	}
+
+	w.state.SetResourceInstanceCurrent(absAddr, src, absProviderConf)
+
+	w.Config[key] = r
+
+	return nil
+}
+
+// Has checks if the given key it's already present or not
+func (w *Writer) Has(key string) (bool, error) {
+	_, ok := w.Config[key]
+	return ok, nil
+}
```

Comment:
I'm not sure this `Has` method or the other one are used anywhere

Reply:
https://github.com/cycloidio/terracognita/pull/33/files#diff-9a4396fc8b469f87eb63a29771b3380fR330 also on the HCL.

The `resource.go` was `collapsed` on my diff maybe you forgot to review it.